db/seqno_to_time_mapping.h

```diff
@@ -265,4 +265,28 @@ class SeqnoToTimeMapping {
   pair_const_iterator FindGreaterEqSeqno(SequenceNumber seqno) const;
 };
 
+// === Utility methods used for TimedPut === //
+
+// Pack a value Slice and a unix write time into buffer `buf` and return a Slice
+// for the packed value backed by `buf`.
+Slice PackValueAndWriteTime(const Slice& value, uint64_t unix_write_time,
+                            std::string* buf);
+
+// Pack a value Slice and a sequence number into buffer `buf` and return a Slice
+// for the packed value backed by `buf`.
+Slice PackValueAndSeqno(const Slice& value, SequenceNumber seqno,
+                        std::string* buf);
+
+// Parse a packed value to get the value and the write time. The unpacked value
+// Slice is backed up by the same memory backing up `value`.
+std::tuple<Slice, uint64_t> ParsePackedValueWithWriteTime(const Slice& value);
+
+// Parse a packed value to get the value and the sequence number. The unpacked
+// value Slice is backed up by the same memory backing up `value`.
+std::tuple<Slice, SequenceNumber> ParsePackedValueWithSeqno(const Slice& value);
+
+// Parse a packed value to get the value. The unpacked value Slice is backed up
+// by the same memory backing up `value`.
+Slice ParsePackedValueForValue(const Slice& value);
```

Comment:
I'm not sure "PackedValue" is a great long term name for this, but it's fine for now. (I thought maybe "AugmentedValue" but that's more letters.) We can always refactor an internal API.

Reply:
I agree that "pack" is a too general term to use here. Thank you for the detailed notes about internal API change practices. I will try to think of a better name, if this needed change, I will probably do it in a follow up. Since I was using the term in all kinds of places including tests too.